analyzer/src/namespace/types.rs

```diff
@@ -543,6 +600,26 @@ impl FeSized for Tuple {
     }
 }
 
+impl FeSized for Struct {
+    fn size(&self) -> usize {
+        self.fields.values().map(|val| val.size()).sum()
```

Comment:
Since the internal encoding of structs currently pads base types to 32 bytes, this should be `self.fields.len() * 32`.